### DISCUSSION TITLE
Modes Disable, Abs and Velocity.

### DISCUSSION THREAD

**Discussion by Wallat1** [bitcraze/discussions#259] [Q&A] [answered]
Hi.

I would like to know what modeDisable, modeAbs and modeVelocity are used for?, since I have seen that they are used in the PID, Mellinger and INDI controllers. On the other hand, I would also like to know if is its use necessary?.

I would appreciate your help, regards.

**Comment by knmcguire** [comment, marked as answer]
Hi! 

Yes these modes use is necessary. It is a way to regulate [the setpoint structure](https://github.com/bitcraze/crazyflie-firmware/blob/681bb85b417b0c2b904735926a256c761446fefb/src/modules/interface/stabilizer_types.h#L190-L211) in the controllers. The setpoint has reference values for position, velocity, attitude and attitude rate. For PID at least, it can not control all of those (unlike mellinger which is a full state controller), so the setpoint should indicate which value to listen to.

So if the [x axis in the setpoint ](https://github.com/bitcraze/crazyflie-firmware/blob/681bb85b417b0c2b904735926a256c761446fefb/src/modules/interface/stabilizer_types.h#L203)is set to `modeAbs`, the controller will use the [position ](https://github.com/bitcraze/crazyflie-firmware/blob/681bb85b417b0c2b904735926a256c761446fefb/src/modules/interface/stabilizer_types.h#L197)as reference, and if it is `modeVelocity`, it will look at [velocity for reference ](https://github.com/bitcraze/crazyflie-firmware/blob/681bb85b417b0c2b904735926a256c761446fefb/src/modules/interface/stabilizer_types.h#L198), and if it is `modeDisabled`, it will not control anything in the x-axis (position nor velocity) and control the attitude instead instead. 

This is an interesting issue for you to read: https://github.com/bitcraze/crazyflie-firmware/issues/577
And here is some documentation: https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/functional-areas/sensor-to-control/commanders_setpoints/#setpoint-structure

**Reply by Wallat1** [reply]
Hi.

Thank you so much for your help, with this information I understood a little more about the modes.

Regards
